### PR TITLE
feat(modals): Call onClose callback when closeModal prop is called

### PR DESCRIPTION
**Pull request by evanpurkhiser**
The issue is we're inconsistent right now, it's called when you click outside of the modal, but now when you trigger the close action of the modal.

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/modalStore.tsx [hunk 1]
```diff
@@ -25,6 +25,13 @@ const ModalStore = Reflux.createStore({
   },
 
   onCloseModal() {
+    const onClose = this.state?.options?.onClose;
+
+    // Trigger the options.onClose callback
+    if (typeof onClose === 'function') {
+      onClose();
+    }
```
Comment (billyvg) [inline src/sentry/static/sentry/app/stores/modalStore.tsx:33]: This doesn't feel great, but neither does using bootstrap modals.

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/stores/modalStore.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/stores/modalStore.tsx:33]: The issue is we're inconsistent right now, it's called when you click outside of the modal, but now when you trigger the close action of the modal.

Comment (billyvg) [reply src/sentry/static/sentry/app/stores/modalStore.tsx:33]: Seems like good info to have in the PR body.

**Comment by matejminar** [comment]
@EvanPurkhiser 
https://github.com/getsentry/sentry/blob/328b261430a14c612731666bc7f5584a9a74f884/src/sentry/static/sentry/app/views/releasesV2/detail/pickProjectToContinue.tsx#L27-L46
This is now broken by this PR. Is there any other way I can run code basically on modal dismiss?

**Pull request merged**